### DISCUSSION TITLE
new version v59 - beta

### DISCUSSION THREAD

**Discussion by nliaudat** [nliaudat/esp32_8ch_motor_shield#59] [Announcements]
Hi, 
After a lot of work, I have draft a new version of the board. 

The major changes are : 

**Better BEMF :** 

- Complete new concept of sensing with no more troubles with resistors to GND
![image](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/6782613/1f9d4dae-72f7-40fd-9f72-c171dd7b797c)

**Remove of AC input :** 

- That was dangerous 
- The cost of HKL-5M was 3.8$ instead of 0.5$ for the new XL1509 module
- 
![image](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/6782613/23913669-8f33-4bfc-9dd1-bfa660681a80)
- With XL1509 module, You can adjust power of motors between 3.3V to 37V (recommended max 12V as not tested)
- Everybody has an old AC/DC 1A charger
- JLCPCB was charging a lot of shipping with the wheight of HKL module
- New DC power input : DC Power Receptacle 2mm 6.4mm + Screw terminal
- 
![image](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/6782613/d1569e20-54ec-4951-beca-4f328dd3e3a5)

**Motor DC  :** 

- input can be 3.3V to 12V 
- a new AMS1117-3.3 powers the ESP32 board

**Better PCB :** 

- Separate 3.3V power of ESP32 board from motors
- Capacitor added to regulate peaks in power
- Via shielding
- Power isolation on bottom side
- Smaller design (126x70mm instead of 126x80). The M4 holes are compatible. 
- Lower the price of all components (jlcpcb price soldered)
- use R0402 footprint if possible 

**ESP32-C3  :** 

-  ESP32-C3 supermini footprint added 
-  ESP32-C3 costs 2.4$ instead of 3.8$ for nodemcu devkit

**OneWire :** 

- Add 3 onewire XH2.54-3P connectors
- Can get input temperature and output temperature of the heating circuit. 

**Studies :** 

- The XL9535 shift register cost the same as 2x SN74HC595, but the esphome code is very new and move to I2C is not compatible with esp-c3 supermini
- The Mini L298N module instead of L9110 soldered is approx the same price (0.7per channel out of 0.52 + resistors)
- Add more channels : if 12, this needs of another opam

**V59 :** 
![image](https://github.com/nliaudat/esp32_8ch_motor_shield/assets/6782613/039c426b-4bf6-4e37-b14f-26c4b7f79df9)

**Comment by Birkemosen** [comment]
Hi, I found your repo, and this looks really interesting.

I would like to test it, since i have 2x6 heating circuits that I would like to optimize for using with an air-to-water heat pump, (having as much water as possible flowing in the system, while keeping the forward temperature as low as possible).

Is the schematic for the prerelease version ready to be shared, or are there issues with it ? (Can’t find them under the prerelease sources)

Thanks in advance

**Reply by nliaudat** [reply]
Hi, 
The XL1509 power module has issue. It cannot run the motors. I have to find another one.

**Reply by nstadigs** [reply]
Is there a problem with the XL1509 module overall or is it your particular copy that's broken?

**Reply by nliaudat** [reply]
I have 10 pieces of XL1509 with all the same result : the motor do not start. 
I mean it's an inductance related issue or a XL1509 internal current limiting protection.

**Comment by nliaudat** [comment]
The problems are corrected. 
I've added a large capacitor and a Schottky diod to prevent esp32 brownout
The XL1509 has enough power but peaks can shutdown esp32
I've bought another 5 pieces of pcb for testing the new version
Please wait till approved. 
Regards

**Reply by nliaudat** [reply]
![image](https://github.com/user-attachments/assets/1fccbf09-ec4d-4d26-85ea-db567ac4272e)

- The AMS1117-3.3 is pcb ready but not in BOM. It's only needed in case if you want to power > 3.3V motors
- Usb-C slot added with 5.1k resitors to get up to 3A power. It's power source only

**Reply by nstadigs** [reply]
And you added usb-c! ❤️

**Comment by nliaudat** [comment]
Some bad news : 

The L9110S chip I use for it's low price is not the same as L9110 and many datasheets are wrong. 
In fact, the S version as no flyback diodes and can support 2.2~6.5V at 100mA  (max 180 mA).

With no incorporated diodes, the proposed BEMF sensing do not work at all. 

I will switch the L9110S module to [TC118S](https://www.lcsc.com/product-detail/Brushed-DC-Motor-Drivers_Shenzhen-Fuman-Elec-TC118S_C88308.html)

**Reply by bezigebever** [reply]
Great work ! So we have to wait a bit for the new board becomes stable and tested ?

**Reply by nliaudat** [reply]
I wait on the new motor driver module ordered at jlcpcb to validate the BEMF.
Then the pcb has to be rewritten

![image](https://github.com/user-attachments/assets/aea18fcd-0ddd-4bc4-9cc7-518ece3102c7)

**Reply by FreezedBrain** [reply]
I am also waiting for the result and the new BOM and Gerber files. The ones from version v58_rev_1.52 are not orderable on jlcpcb because parts are missing in the BOM. 

Error:
The below parts won't be assembled due to data missing.
WROOM-LEFT1,WROOM-RIGHT,WROOM-LEFT,HLK-PM03,P1,P2,P3,P4,H1,H3,H4,H5,OPAMP-H1,U1 designators don't exist in the BOM file.

**Reply by nliaudat** [reply]
It will take more time. 
I received 10x T118s modules I ordered with the previous schematic. 

They are not working at all and overheating. 
I search the problem without success

If I put 3.3V on INA or INB, the module get warm and the power on VCC drop to 1.6v
I tried to add 1k resistors to lower INx cause their limit is 20mA = fail. 

I have to test others : 
But there is no documentation on BEMF for all of them. 
In fact, getting BEMF from GND as in L9110S was a luck, but the L9110S module is end of life and less available.

![image](https://github.com/user-attachments/assets/102a0f42-99c8-4d03-ab53-0ef60767b3d8)

**Comment by nliaudat** [comment]
feel free to test or adapt yourself. 

Here is the easyeda pro project
[ReuseBlock_TC118s_2024-10-09.zip](https://github.com/user-attachments/files/17315357/ReuseBlock_TC118s_2024-10-09.zip)

**Comment by nliaudat** [comment]
I have good news : 

I found the problem with T118s module. (shortcut in OA / OB with GND)
The 2nd problem was the diodes D1/D2, the motor need to be grounded in every case.  It was a stupid idea.

With the new design, there is no need for opamp as the output in running conditions is 0.7V and 0.8 at stall,

The new diode D4 between BEMF and GND increase accuracy from 0.2v to 0.7V in running state. 
The fact of injecting BEMF in GND was disturbing ADC reading. 

![image](https://github.com/user-attachments/assets/62023520-b360-409d-a5a1-6a9850c59972)

**Comment by nliaudat** [comment]
The final board ordered today : 

![image](https://github.com/user-attachments/assets/bd0826c6-9d8e-43cd-8949-26a352923f66)

The power is usb-c only cause it's enough.

**Reply by nstadigs** [reply]
Very nice! Glad you solved it. It also got smaller!

Few reflections looking at the image:
- Shouldn't the usb port be at the edge of the board? You never know how thick the holster for the connector is on the cable.
- Lower right screw hole seems hard to use if you soldered the connector.

**Reply by nliaudat** [reply]
Hi, 
Many thanks for you comments. 

I will make a hole for usb-c holder and verify the screw hole. 
Regards

**Reply by FreezedBrain** [reply]
Keep your fingers crossed that everything works. Many thanks for your efforts.

**Reply by bezigebever** [reply]
Great job ! If you do final testing I'll defenitely order these boards ;-)

**Reply by marcelveldt** [reply]
If you think its ready for more folks to test it, I'd happily test it out as well if you have the final files.

**Reply by nliaudat** [reply]
I mean the new version needs to be reviewed before final release. 
I have many questions about what's your really needs.
I'll post many surveys soon about power supply, esp-32 version, connectors, max price, etc.

**Reply by marcelveldt** [reply]
I'm very interested in this board (either the old or the new revision) - currently it cant be ordered due to some components of the BOM are no longer available.

**Reply by bezigebever** [reply]
@nliaudat what is the status of the board ? I would really like to order some ;-)

**Reply by nliaudat** [reply]
All preliminary tests are carried out. 
I have a bug with the ADC BEMF measurement when it comes at endstop (0.700V). 
This causes the ESP32 to reboot ...
By adding a 1.2k resistor to BEMF, it no longer crashes. 
For the moment, I don't understand why and I'm looking for answers before final publication. 

Another problem is that TC118s output is only 2.6V on motor as it should be 3.3V

**Reply by nliaudat** [reply]
After some more tests, one resistor of low value as 10 ohm solves the reboot problem. 
I mean that it's linked to the filtering property of the resistor which act as a low pass filter or perhaps an impedance reduction. 

Finally, I will add a 180 ohm resistor cause it's suitable and can [correct some nodemcu devkits ADC reading](https://arduino.stackexchange.com/questions/71949/voltage-divider-and-nodemcu-inputs/71952#71952). 

Here are the ADC values : 

- 0.383 V running on pcb only without motor
- 0.610 V opening/closing with no load
- 0.620-0.635 V opening with load
- 0.620-0.635 V closing with load
- 0.650 V endstop

**Reply by bezigebever** [reply]
@nliaudat So you wil make a new version tot test it again ?
I would love to go for this version without the 230Vac connection

**Comment by nliaudat** [comment]
Please have a look at new surveys: https://github.com/nliaudat/esp32_8ch_motor_shield/discussions

**Comment by nliaudat** [comment]
New version 61 rev 4.51 ready : 
![image](https://github.com/user-attachments/assets/c54a13bf-4a17-435d-86ea-ec8a5355aaf0)

**Reply by nliaudat** [reply]
## PCB sharing
[https://oshwlab.com/nliaudat/floor_heatin_controller_v61](https://pro.easyeda.com/editor#id=15bc77acaa4147fca2a36e92e8cc8681)

**Reply by bezigebever** [reply]
@nliaudat many thanks for your effort !
This one is 'production ready' ?

**Reply by nliaudat** [reply]
I have debug everythings with the v61 beta version.
That version should works,
I'll receive the final pcb in 2 weeks. 
If you feel not sure, please wait until. 
Regards

**Reply by bezigebever** [reply]
@nliaudat i'll take the risk ;-)
I suppose best is order directly, I'm JLPCB client...
I uploaded gerber, BOM and CPL, now get this error:
```
The below parts won't be assembled due to data missing.
supermini-1,WROOM-LEFT2,5-12V-input,H1,WROOM-LEFT1,WROOM-RIGHT1 designators don't exist in the BOM file.
```

**Reply by bezigebever** [reply]
And BOOMELE(Boom Precision Elec) 2.0-1*5P is not available (P1-P8)
Are these needed ?

**Reply by nliaudat** [reply]
No, all that stuff are not in the BOM. 
It's only used for pcb footprints

**Reply by bezigebever** [reply]
@nliaudat so how do I order an assembled PCB then ?

**Reply by nliaudat** [reply]
I have ordered 5 boards sunday with no problem. You can just ignore that parts.

**Reply by THorst92** [reply]
@nliaudat Do you know why the cut outs are not working when I want to order the PCB:
<img width="989" alt="image" src="https://github.com/user-attachments/assets/fe396287-2fd2-49a2-9e90-9b99a1b3f797">

**Reply by nliaudat** [reply]
Are you ordering at jlcpcb ? 
Have you done modifications on gerber files ? 
Can you try to pass your upload to https://jlcdfm.com/

**Reply by THorst92** [reply]
I directly ordered from the link above ([https://oshwlab.com/nliaudat/floor_heatin_controller_v61](https://pro.easyeda.com/editor#id=15bc77acaa4147fca2a36e92e8cc8681))
I also tried to download the Gerber files (Exported from EasyEDA and taken from the release) and upload it again without any modifications.
https://jlcdfm.com/ gives me the same result :( 
When I select the 3D-view in EasyEDA the cutouts are displayed correctly.

**Reply by nliaudat** [reply]
Can you try that ? 
This is the production file generated by jlcpbc after validation on my last order : 
[Gerber_61-451_2024-11-17_Y61.zip](https://github.com/user-attachments/files/17950900/Gerber_61-451_2024-11-17_Y61.zip)

**Reply by THorst92** [reply]
This is working! Thanks a lot! :)

**Reply by grericht** [reply]
i want to order 5 pcs but i get unselected parts:
C35,C36 | 470uF | CAP-SMD_BD6.3-L6.6-W6.6-FD |   | 2
P1,P2,P3,P4,P5,P6,P7,P8 | 2.0-1*5P母 | HDR-TH_5P-P2.00-V-F-1 |   | 8
SCREW1,SCREW2,SCREW3,SCREW4 | Screw-Hole-M2 | Screw-Hole-M2 |   | 4

i dont need them? C35 and C36?

**Comment by nliaudat** [comment]
I received the new boards and they works as expected. :)

**Reply by WMP** [reply]
Do you have 2 to sell to Poland?

**Reply by THorst92** [reply]
@nliaudat @bezigebever 
Do you have one PCB (version 61 rev 4.51) which you could sell me (Germany) so that I do not have to order 5 pieces?

**Reply by marcelveldt** [reply]
I just ordered 5 pieces (need 2 myself) - I'm in the Netherlands. Contact me if you want one
EDIT: sold

**Reply by nliaudat** [reply]
Please open a new buy/sell discussion

**Comment by nliaudat** [comment]
v61 completed

**Comment by nliaudat** [comment]
For v61, plz update your firmware to latest version as IA/IB motor control is inverted
https://github.com/nliaudat/floor-heating-controller/releases
